tests/filecheck/dialects/riscv/frep_verification.mlir

```diff
@@ -36,3 +35,40 @@ riscv_snitch.frep_outer %i0 {
 }
 
 // CHECK: Operation does not verify: Frep operation body may not contain instructions with side-effects, found riscv.sw
+
```

Comment:
I don't think we need any of these tests; they just replicate the trait's expected behaviour which has already been tested.

Reply:
I've been thinking about this, and I'm not sure still one way or the other. The behaviour of the op is distinct from the implementation, so they are decouple from whether a trait is used or custom code like I had in an earlier state of this PR. I think that it makes sense for something to fail if we remove the implicit terminator trait from `FrepOperation`. But then maybe that's overly defensive.

Reply:
Yeah, I don't know, we haven't been doing in general AFAIU.
Happy to keep it if your think the API is still in flux here.